x-pack/plugins/security_solution/public/detections/containers/detection_engine/alerts/use_host_isolation.tsx

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { useCallback, useState } from 'react';
+import { errorToToaster, useStateToaster } from '../../../../common/components/toasters';
+import { isSecurityAppError } from '../../../../common/utils/api';
+import { createHostIsolation } from './api';
+
+interface HostIsolationStatus {
+  loading: boolean;
+  isolateHost: () => Promise<boolean>;
+}
+
+interface UseHostIsolationProps {
+  agentId: string;
+  comment: string;
+}
+
+export const useHostIsolation = ({
+  agentId,
+  comment,
+}: UseHostIsolationProps): HostIsolationStatus => {
+  const [loading, setLoading] = useState(false);
+  const [, dispatchToaster] = useStateToaster();
+
+  const isolateHost = useCallback(async () => {
+    let isSubscribed = true;
```

Comment:
I am not seeing the purpose of this variable (I seem to remember us talking about this) and it just seems to be adding local complexity that you really don't need. I think this callback can be shorten to: 

```typescript
const isolateHost = useCallback(async () => {
  try {
    setLoading(true);
    const isolationStatus = await createHostIsolation({ agentId, comment });

    setLoading(false);
    return isolationStatus.action ? true : false;
  } catch (error) {
    
    // 500 error
    if (isSecurityAppError(error) && error.body.status_code !== 404) {
      errorToToaster({
        title: i18n.translate('xpack.securitySolution.hostIsolation.apiError', {
          defaultMessage: 'Stuff went wrong',
        }),
        error,
        dispatchToaster,
      });
    }
  }
  
  return false;
  
}, [agentId, comment, dispatchToaster]);

```